src/pubsub.c

```diff
@@ -306,31 +484,39 @@ int pubsubPublishMessage(robj *channel, robj *message) {
             receivers++;
         }
     }
-    /* Send to clients listening to matching channels */
-    di = dictGetIterator(server.pubsub_patterns);
-    if (di) {
-        channel = getDecodedObject(channel);
-        while((de = dictNext(di)) != NULL) {
-            robj *pattern = dictGetKey(de);
-            list *clients = dictGetVal(de);
-            if (!stringmatchlen((char*)pattern->ptr,
-                                sdslen(pattern->ptr),
-                                (char*)channel->ptr,
-                                sdslen(channel->ptr),0)) continue;
-
-            listRewind(clients,&li);
-            while ((ln = listNext(&li)) != NULL) {
-                client *c = listNodeValue(ln);
-                addReplyPubsubPatMessage(c,pattern,channel,message);
-                receivers++;
+
+    if (!meta.local) {
```

Comment:
Would be good to document why? I would also do:
```
if (meta.local) {
    /* Local pubsub ignores patterns */
    return receivers;
}
```
to reduce indentation and make the flow clearer.